# Calculate speckle signature score
This pipeline describes how to calculate speckle signature scores from RNA-seq data. In this example, it is 10 cancer or normal kidney samples.

# Expression data
This analysis starts with a table of normalized counts. For example, a normalized counts file generated by Deseq2. In this example ("normalizedCounts_10samples.txt"), genes are rows and samples are columns.

# Calculate speckle scores
The following script reads the files "normalizedCounts_10samples.txt", "speckleProteinGenes_22cancers.txt" (all speckle protein genes within the signature), "sigI_speckleProteinGenes_22cancers.txt", and "sigII_speckleProteinGenes_22cancers.txt" to generate speckle signature scores based on gene expression. It calculates scores based on z-scores of the data, so the speckle score will vary depending on which samples are included and thus should be viewed as a relative rather than absolute quantification of speckle signature. 

```Rscript makeSpeckleSignatureHeatmapAndCalculateSpeckleScore.R```

The above script generates two files:
1. "speckleScores.txt" - A list of speckle scores for each sample
2. "speckleSignatureHeatmap_PDX.pdf" - A heatmap of expression of speckle protein genes with the rows sorted by speckle score and the columns hierachically clustered based on expression. Speckle protein genes are annotated at the top of the heatmap. If the sample group shows the cononical speckle signature, Signature I speckle protein genes should roughly cluster together and away from Signature II speckle protein genes.








